src/main/scala/scalismo/statisticalmodel/DiscreteGaussianProcess.scala

```diff
@@ -63,13 +63,13 @@ class DiscreteGaussianProcess[D <: Dim: NDSpace, Value] private[scalismo] (val m
    * The marginal distribution for the points specified by the given point ids.
    * Note that this is again a DiscreteGaussianProcess.
    */
-  def marginal(pointIds: Seq[PointId])(implicit domainCreator: UnstructuredPointsDomain.Create[D]): DiscreteGaussianProcess[D, Value] = {
+  def marginal(pointIds: Seq[PointId])(implicit domainCreator: UnstructuredPointsDomain.Create[D]): DiscreteGaussianProcess[D, UnstructuredPointsDomain[D], Value] = {
```

Comment:
It is at least not straight-forward to implement. As the GP is already discrete we cannot sample it at arbitrary domain points. So if the GP represents for example a ```UnstructuredPointsDomain```, it is not possible to resample it with a ```DiscreteImageDomain```. Encoding this in types seems not worth the trouble. But we can achieve the same functionality by first interpolating and then discretizing again.